flux-kube/plugins/flux-kube.c

```diff
@@ -0,0 +1,352 @@
+
+    if (!(h = flux_jobtap_get_flux (p))
+        || flux_job_kvs_key (key, sizeof (key), id, "R") < 0
+        || !(txn = flux_kvs_txn_create ())
+        || flux_kvs_txn_put (txn, 0, key, R) < 0
+        || !(f = flux_kvs_commit (h, NULL, 0, txn))
+        || flux_future_then (f, -1, alloc_continuation, p)
+        || !(idptr = calloc (1, sizeof (*idptr)))
+        || flux_future_aux_set (f, "jobid", idptr, free) < 0) {
+        flux_kvs_txn_destroy (txn);
+        flux_future_destroy (f);
+        free (idptr);
+        return -1;
+    }
+    *idptr = id;
+    flux_kvs_txn_destroy (txn);
+    return 0;
+}
+
+
+static int sched_cb (flux_plugin_t *p,
+                     const char *topic,
+                     flux_plugin_arg_t *args,
+                     void *arg)
+{
+    const char *R = NULL, *jobspec = NULL;
+    flux_jobid_t id;
+    flux_future_t *f;
+    int retval = -1;
+
+    /*  If flux-kube::R set on this job then commit R to KVS,
+     *  get the jobspec, submit the jobspec to flux-kube.py, 
+     *  and set flux-kube flag.
+     */
+    if (!(R = flux_jobtap_job_aux_get (p,
+                                       FLUX_JOBTAP_CURRENT_JOB,
+                                       "flux-kube::R")))
+        return 0;
+
+    if (!(jobspec = flux_jobtap_job_aux_get (p,
+                                             FLUX_JOBTAP_CURRENT_JOB,
+                                             "flux-kube::jobspec")))
+        return 0;
+
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:I}",
+                                "id", &id) < 0) {
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                     "alloc", 0,
+                                     "flux-kube: %s: unpack: %s",
+                                     topic,
+                                     flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
+    if (!(f = flux_rpc_pack (flux_jobtap_get_flux (p), "fluxkube.submit", 
+                             FLUX_NODEID_ANY, 0,
+                             "{s:s}", "jobspec", jobspec))) { 
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                     "alloc", 0,
+                                     "flux-kube: %s: Failed to send RPC: %s",
+                                     topic,
+                                     strerror (errno));
+        return -1;
+    }
+    
+    if (flux_future_then (f, -1.0f, get_response_continuation, p) < 0) {
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                     "alloc", 0,
+                                     "flux-kube: %s: Failed to "
+                                     "submit jobspec: %s",
+                                     topic,
+                                     strerror (errno));
+        return -1;
+    }
+
+    if (alloc_start (p, id, R) < 0) {
```

Comment:
Here's a question I didn't consider before -- should the post of the `alloc` event wait until after `fluxkube.submit` returns successfully? Sorry if that is a naive question, I'm not sure I have a clear picture on how the Flux job being handled here participates with the `fluxkube.submit` request.

Reply:
Good question.  My understanding is that it shouldn't matter because if `fluxkube.submit` fails it will return -1 in the `sched_cb` and initiate the `cleanup_cb`.

Reply:
I will address this in a future PR if the behavior isn't as I expect.